src/main/java/domain/entities/taskmanager/Event.java

```diff
@@ -0,0 +1,84 @@
+package domain.entities.taskmanager;
+
+import core.exceptions.InvalidArgumentException;
+import core.singletons.Singletons;
+import core.utils.Pair;
+import core.utils.TokenUtilities;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+import java.util.Map;
+import java.util.Set;
+
+/**
+ * An Event is a Task with a starting time and a ending time.
+ */
+public class Event extends Task {
+    /**
+     * The token for getting the start time.
+     */
+    private static final String startAtKey = "/from";
+    /**
+     * The token for getting the end time.
+     */
+    private static final String endAtKey = "/to";
+    /**
+     * The set for retrieval of the values from the Token Utils
+     */
+    private static final Set<String> delims = Set.of(startAtKey, endAtKey,
+            Task.completeKey);
+    private final LocalDate startAt;
+    private final LocalDate endAt;
+
+    /**
+     * Creates a new Event object from the tokens.
+     *
+     * @param tokens The keywords after which we retrieve the important
+     *               information.
+     * @throws InvalidArgumentException for an event object to be valid,
+     *                                  its name must not be null, and it must
+     *                                  have a starting time and an ending time.
+     */
+    public Event(String[] tokens) throws InvalidArgumentException {
```

Comment:
Code is a little difficult to follow here, particularly the predicates in your if and else if statements. A suggestion is to define them as variables with proper naming beforehand so the code is more easy to follow! 😃